cockpit-plugin/src/YamlFile.jsx

```diff
@@ -0,0 +1,124 @@
+import cockpit from 'cockpit';
+import React from 'react';
+import { TextArea, Button, Alert, Split, SplitItem, Stack, StackItem } from "@patternfly/react-core";
+const JSYaml = require("js-yaml");
+
+export class YamlFile extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { content: "", modified: false, hasError: false, errMessage: "" };
+
+        this.loadContent();
+    }
+
+    updateState = (c, m = false, he = false, em = "No error") => {
+        this.setState({ content: c, modified: m, hasError: he, errMessage: em });
+    };
+
+    loadContent = () => {
+        // Check cockpit.channel for permission elevation
+        // https://cockpit-project.org/guide/latest/cockpit-channels.html#cockpit-channels-channel
+        let fileName = this.props.fileName;
+
+        const promise = cockpit.file(fileName).read();
+        promise.then((content, tag) => {
+            if (!content) {
+                fileName += ".default";
+                const promise = cockpit.file(fileName).read();
+                promise.then((content, tag) => {
+                    if (content) {
+                        this.updateState(content);
+                        this.validateContent();
+                    } else {
+                        this.updateState("", false, true, "Failed to open " + this.props.fileName);
```

Comment:
This notifies about failing to read `${filename}.default`, I think include of suffix would be more accurate and helpful in case of error